### PR TITLE
Support delay notifications

### LINKED ISSUE #8: [$50] Support reminders - delayed notifications
Please create a new x-delayed exchange`connect-notifications-reminders`, that can publish a msg with routing key. Eg. project.copilot-unclaimed for our current use case. `connect-notifications` queue can subscribe to that routing key as well. 

That way reminders are local to connect-notifications-service and don't impact other consumers that subscribe to the same routing key.

Testing: 
Please add the necessary tests. Also this needs to work in CircleCI so you might need to setup / enable rabbitmq server with delayed msg plugin.

### PR DESCRIPTION
**Pull request by uttpal**
fix #8

### CODE REVIEW

File: config/development.json [hunk 1]
```diff
@@ -4,9 +4,13 @@
   "SOURCE_RABBIT_EXCHANGE_NAME": "dev.projects",
   "SOURCE_RABBIT_QUEUE_NAME": "dev.connect-notifications",
   "TARGET_RABBIT_EXCHANGE_NAME": "dev.notifications",
+  "TARGET_RABBIT_DELAY_EXCHANGE_NAME": "dev.connect-notifications-reminders",
+  "TARGET_RABBIT_DELAY_ROUTING_KEY": "project.copilot-unclaimed",
```
Comment (parthshah) [inline config/development.json]: Please move TARGET_RABBIT_DELAY_ROUTING_KEY from development to default config. Only keys that need to be overwritten should be added to development, production or test config files.

File: config/development.json [hunk 2]
```diff
@@ -4,9 +4,13 @@
...
   "TARGET_RABBIT_QUEUE_NAME": "dev.notifications",
 
   "COPILOT_TARGET_RABBIT_QUEUE_NAME": "dev.notifications-slack-copilot",
+  "COPILOT_TARGET_RABBIT_ROUTING_KEY": "slack.copilot",
```
Comment (parthshah) [inline config/development.json]: Also move to default.config

File: config/development.json [hunk 3]
```diff
@@ -4,9 +4,13 @@
...
+  "MANAGER_TARGET_RABBIT_ROUTING_KEY": "slack.manager",
```
Comment (parthshah) [inline config/development.json]: Move to default.config

File: config/production.json [hunk 4]
```diff
@@ -4,9 +4,13 @@
   "SOURCE_RABBIT_EXCHANGE_NAME": "projects",
   "SOURCE_RABBIT_QUEUE_NAME": "connect-notifications",
   "TARGET_RABBIT_EXCHANGE_NAME": "notifications",
+  "TARGET_RABBIT_DELAY_EXCHANGE_NAME": "dev.connect-notifications-reminders",
+  "TARGET_RABBIT_DELAY_ROUTING_KEY": "project.copilot-unclaimed",
```
Comment (parthshah) [inline config/production.json]: Same as above, please remove. We just need this in default config.

File: config/production.json [hunk 5]
```diff
@@ -4,9 +4,13 @@
...
   "TARGET_RABBIT_QUEUE_NAME": "notifications",
 
   "COPILOT_TARGET_RABBIT_QUEUE_NAME": "notifications-slack-copilot",
+  "COPILOT_TARGET_RABBIT_ROUTING_KEY": "slack.copilot",
```
Comment (parthshah) [inline config/production.json]: remove

File: config/production.json [hunk 6]
```diff
@@ -4,9 +4,13 @@
...
+  "MANAGER_TARGET_RABBIT_ROUTING_KEY": "slack.manager",
```
Comment (parthshah) [inline config/production.json]: remove

File: config/test.js [hunk 7]
```diff
@@ -22,8 +22,11 @@ module.exports = {
 
   COPILOT_TARGET_RABBIT_QUEUE_NAME: 'test.notifications-slack-copilot',
   MANAGER_TARGET_RABBIT_QUEUE_NAME: 'test.notifications-slack-manager',
+  COPILOT_TARGET_RABBIT_ROUTING_KEY: 'slack.copilot',
```
Comment (parthshah) [inline config/test.js]: remove both copilot, manager & target delay routing keys

File: src/handlers/util.js [hunk 8]
```diff
@@ -22,15 +22,16 @@ function requestPromise(options, cb = null) {
   // setting default options
   _.defaults(options, { method: 'GET', json: true })
   return new Promise((resolve, reject) => {
-    request(options, (err, res, data) => {
+    request[options.method.toLowerCase()](options, (err, res, body) => {
```
Comment (parthshah) [inline src/handlers/util.js]: Having method property in options and invoking that specific function on request body are redundant, right?
`request ({method: GET, url: ... })` and `request.get({url: ... })` are effectively the same thing.

File: src/handlers/util.js [hunk 9]
```diff
@@ -22,15 +22,16 @@ function requestPromise(options, cb = null) {
...
+      const data = JSON.parse(body);
```
Comment (parthshah) [inline src/handlers/util.js]: By passing `json:true` in the options body has the same effect?
https://github.com/request/request#requestoptions-callback

Comment (uttpal) [inline src/handlers/util.js]: I did this to stub requests easily

Comment (parthshah) [inline src/handlers/util.js]: Ah okay. Mind adding a comment in there so we don't refactor this code and break the tests.

**Comment by uttpal** [comment]
@parthshah changes done, circle ci tests passing failed while aws deployment ( aws keys not added)

**Pull request merged**